packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -198,6 +199,16 @@ class Dart2JSTarget extends Target {
     'dart2js.d',
   ];
 
+  String _collectOutput(ProcessResult result) {
```

Comment:
Nice refactor here.

Reply:
Well, yeah, and the original code wasn't taking into account that stdout and stderr can be either `String` or `List<int>` types. They're almost certainly `String`s, but still.

I wish we were using `process_runner` instead of `process` here, so we could have actual interleaved combined output, but this works for now.